### PR TITLE
1159 - cloudevent support for kfserving

### PR DESCRIPTION
**Pull request by mtickoobb**
**What this PR does / why we need it**:
Support cloudevent for kfserving
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1159

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: python/kfserving/kfserving/kfmodel.py [hunk 1]
```diff
@@ -51,7 +51,16 @@ def load(self) -> bool:
         return self.ready
 
     def preprocess(self, request: Dict) -> Dict:
-        return request
+        # If cloudevent dict, then parse 'data' field. Otherwise, pass through.
+        if  "ce-data" in request
+        and "ce-time" in request
```
Comment (yuzisun) [inline python/kfserving/kfserving/kfmodel.py:56]: I think these names are http header names, when decoded as python dict the names are without `ce-` prefix

File: python/kfserving/kfserving/handlers/http.py [hunk 2]
```diff
@@ -43,21 +44,38 @@ def validate(self, request):
...
         request = model.preprocess(body)
         request = self.validate(request)
         response = (await model.predict(request)) if inspect.iscoroutinefunction(model.predict) else model.predict(request)
         response = model.postprocess(response)
+        if self.has_ce_headers(self.request.headers):
+            event = CloudEvent(self.request.headers, response)
+            if is_binary(self.request.headers):
+                header, response = to_binary(event)
+            elif is_structured(self.request.headers):
+                header, response = to_structured(event)
```
Comment (yuzisun) [inline python/kfserving/kfserving/handlers/http.py:78]: you might need to call `self.set_header(header)` before `self.write`

File: python/kfserving/test/test_server.py [hunk 3]
```diff
@@ -141,3 +183,32 @@ async def test_model_not_ready_error(self, http_server_client):
...
+        event = dummy_cloud_event()
+        headers, body = to_binary(event)
+        resp = await http_server_client.fetch('/v1/models/TestModel:predict',
+                                              method="POST",
+                                              headers=headers,
+                                              body=body)
+
+        assert resp.code == 200
+        assert resp.body == b'{"predictions": [[1, 2]]}'
+        assert resp.headers['content-type'] == "application/x-www-form-urlencoded"
```
Comment (yuzisun) [inline python/kfserving/test/test_server.py:206]: is this expected?

File: python/kfserving/test/test_server.py [hunk 4]
```diff
@@ -49,6 +66,19 @@ async def load(self, name: str) -> bool:
         else:
             return False
 
+def dummy_cloud_event():
```
Comment (yuzisun) [inline python/kfserving/test/test_server.py:69]: Add a test for binary data also?

File: python/kfserving/test/test_server.py (hunk 3 above)
Comment (mtickoobb) [reply python/kfserving/test/test_server.py:206]: Ya I think this is the content-type for binary http post messages

**Pull request merged**